plugin/restore/mysql/mysql.go

```diff
@@ -0,0 +1,121 @@
+package mysql
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/bytebase/bytebase/plugin/db/mysql"
+)
+
+// BinlogConfig is the binlog coordination for MySQL.
+type BinlogConfig struct {
+	Filename string
+	Position int64
+}
+
+// Restore implements recovery functions for MySQL
+type Restore struct {
+	driver *mysql.Driver
+}
+
+// New creates a new instance of Restore
+func New(driver *mysql.Driver) *Restore {
+	return &Restore{
+		driver: driver,
+	}
+}
+
+// RestoreBinlog restores the database using incremental backup in time range of [config.Start, config.End).
+func (r *Restore) RestoreBinlog(ctx context.Context, config BinlogConfig) error {
+	return fmt.Errorf("Unimplemented")
+}
+
+// RestorePITR is a wrapper for restore a full backup and a range of incremental backup
+func (r *Restore) RestorePITR(ctx context.Context, fullBackup *bufio.Scanner, config BinlogConfig, database string, timestamp int64) error {
+	pitrDatabaseName := getPITRDatabaseName(database, timestamp)
+	query := fmt.Sprintf("CREATE DATABASE `%s`; USE `%s`; SET foreign_key_checks=OFF", pitrDatabaseName, pitrDatabaseName)
```

Comment:
Use multiple lines so that it's easier to read.
Add a comment on why we turn off foreign_key_checks and when we should turn it back on, or maybe we don't have to if that's per-session.